certification/internal/shell/has_unique_tag.go

```diff
@@ -13,31 +11,40 @@ import (
 type HasUniqueTagCheck struct{}
 
 func (p *HasUniqueTagCheck) Validate(image string) (bool, error) {
-	imageName := strings.Split(image, ":")[0]
-	stdouterr, err := exec.Command("skopeo", "list-tags", "docker://"+imageName).CombinedOutput()
+	tags, err := p.getDataToValidate(image)
 	if err != nil {
-		log.Error("unable to execute skopeo on the image: ", err)
 		return false, err
 	}
+	return p.validate(image, tags)
+}
+
+func (p *HasUniqueTagCheck) getDataToValidate(image string) ([]string, error) {
+
+	runReport, err := skopeoEngine.ListTags(image)
 
-	// we must send gojq a []interface{}, so we have to convert our inspect output to that type
-	var skopeoData map[string]interface{}
-	err = json.Unmarshal(stdouterr, &skopeoData)
 	if err != nil {
-		log.Error("unable to parse skopeo list-tags data for image", err)
-		log.Debug("error marshaling skopeo list-tags data: ", err)
-		log.Trace("failure in attempt to convert the raw bytes from `skopeo list-tags` to a [map[string]interface{}")
-		return false, err
+		log.Error("unable to execute skopeo on the image: ", err)
+		return nil, err
 	}
-	tags := skopeoData["Tags"].([]interface{})
 
-	var tagsString string
-	for _, tag := range tags {
-		tagsString = tagsString + tag.(string) + " "
+	rawTags := runReport.Stdout["Tags"].([]interface{})
```

Comment:
You wouldn't need to do all of this if you extract the Tags in the ListTags method.